### PR TITLE
Jeffery/Queue name can be specified in config file

### LINKED ISSUE #106: config.ini cleanup
Right now, the dashboard is reading from a `config.ini` file located in `src`, we should change this so that it reads from a config file inside 'build' instead.

Also, we should rename the current config.ini to config.ini.example, and tell the user to update it in the README. 

Lastly, somewhere along the line the IP address was changed to be our address, let's fix that up to be localhost again.

### PR DESCRIPTION
**Pull request by jeffh1996**
From Bill's comments on my previous pull request

> The issue that would be brought up would be that if multiple instances ran with the same queue_name, they would all take turns getting stuff from the queue, ie it would be like instance A gets the first message, B gets the second message, and so on. What I think we should do is remove the queueName from config.ini, then set the queue_name by the Type - race mode as raceQueue, display mode as displayQueue, and debug mode as debugQueue. That way, three modes won't interfere with each other.

> We should also add another command line argument to specify the queue name ourselves, in case we want to connect to it some other way, like through our laptops.

Queue name is now not dependent on the config.ini file. The default queue names for each queue are raceQueue, displayQueue, and debugQueue with an option to use a custom queue name for each mode by adding an additional flag and the queue name after in the terminal.

Also Fixes #106

### CODE REVIEW

File: src/EpsilonDashboard/EpsilonDashboard.cpp [hunk 1]
```diff
@@ -8,33 +8,50 @@
...
+    }
+
+    QString queueSpecified = parser.value(queueNameOption);
+
+    if (queueSpecified != "")
+    {
+        queueName = queueSpecified;
     }
 
+    infrastructureContainer_.reset(new InfrastructureContainer(queueName));
```
[the remark this answers is not in the document]
Comment (bill-luu) [reply src/EpsilonDashboard/EpsilonDashboard.cpp]: Separate default queue names. Thanks!

File: src/EpsilonDashboard/EpsilonDashboard.cpp [hunk 2]
```diff
@@ -21,17 +24,23 @@ EpsilonDashboard::EpsilonDashboard(int& argc, char** argv)
...
     parser.addOption(raceModeOption);
     parser.addOption(debugModeOption);
+
     parser.process(*this);
     Mode mode = Mode::DISPLAY;
 
     if (parser.isSet(raceModeOption))
     {
         mode = Mode::RACE;
+        infrastructureContainer_->setQueueName("raceQueue");
```
Comment (bill-luu) [inline src/EpsilonDashboard/EpsilonDashboard.cpp]: Let's toss the constant queue names into an include file for use.

**Comment by jeffh1996** [comment]
I tossed the constant queue names into the infrastructure container header file. The reason why I chose this is because both the EpsilonDashboard class and the InfrastructureClass use these constant names so I figured I put them in a place where it is accessible from both classes. Let me know what you think.

**Comment by bill-luu** [comment]
Instead of being a part of a class, I think it would make more sense for them to be in a namespace.

File: src/InfrastructureLayer/InfrastructureContainer.cpp [hunk 3]
```diff
@@ -14,32 +22,15 @@ InfrastructureContainer::~InfrastructureContainer()
 
 void InfrastructureContainer::setQueueName(const char* queueName)
 {
-    if (settings_->queue() == raceQueue_ ||
-            settings_->queue() == displayQueue_ ||
-            settings_->queue() == debugQueue_ ||
-            settings_->queue() == defaultQueue_)
+    if (settings_->queue() == RACE_QUEUE ||
```
Comment (bill-luu) [inline src/InfrastructureLayer/InfrastructureContainer.cpp]: Why is this if statement necessary?

Comment (jeffh1996) [reply src/InfrastructureLayer/InfrastructureContainer.cpp]: The default queue names override the custom queue names. Basically, this detects if we already have a custom queue name in place.

For example, without this statement:
If we put in a custom queue name `customQueue` into the config file, and try to run it in Race mode, The `InfrastructureContainer` is created by reading in the config file and the queue name at this point will be `customQueue`. However, the `EpsilonDashboard` then parses the input arguments recognizes that it is running in Race mode and therefore resets the queue name to be `raceQueue`.

**Pull request merged**